src/test/scala/lms/thirdparty/test_size_t.scala

```diff
@@ -0,0 +1,47 @@
+package lms
+package thirdparty
+
+import lms.core.stub._
+import lms.core.virtualize
+import macros.SourceContext
+import lms.collection._
+
+class SizeTTest extends TutorialFunSuite {
+  val under = "thirdparty/size_t"
+
+  abstract class DslDriverCSizeT[A: Manifest, B: Manifest] extends DslDriverC[A,B] with SizeTOps { q =>
+    override val codegen = new DslGenC with CCodeGenSizeTOps {
+      val IR: q.type = q
+    }
+  }
+
+  test("basic") {
+    val driver = new DslDriverCSizeT[Int, Unit] {
+      @virtualize
+      def snippet(arg: Rep[Int]) = {
+        val f = topFun { a: Rep[SizeT] =>
+          printf("size t is %d", a.toInt)
+        }
+        f(SizeT(5))
```

Comment:
Does `f(5)` work here?

Reply:
it is not working yet because you would need 2 level of implicit conversion to get Rep[SizeT] from Int